src/transformers/adapters/models/bart.py

```diff
@@ -104,6 +105,13 @@ def enable_adapters(self, adapter_names: list, unfreeze_adapters: bool, unfreeze
         self.attention_adapters.enable_adapters(adapter_names, unfreeze_adapters, unfreeze_attention)
         self.output_adapters.enable_adapters(adapter_names, unfreeze_adapters, unfreeze_attention)
 
+    # Makes sure the "parent" reference always points to the correct module.
+    # This is especially relevant when using torch data parallelism.
```

Comment:
Maybe, an adapter test with data parallelism would be useful to detect errors like this.